crates/driver/src/domain/quote.rs

```diff
@@ -134,16 +134,20 @@ impl Order {
         }
     }
 
-    /// The asset being sold, or [`eth::U256::max_value`] if this is a buy, to
+    /// The asset being sold, or a very large value if this is a buy, to
     /// facilitate surplus.
+    ///
+    /// Note that we intentionally do not use [`eth::U256::max_value()`] as an
+    /// order with this would cause overflows with the smart contract, so buy
+    /// orders requiring excessively large sell amounts would not work anyway.
```

Comment:
I wouldn't make this a doc comment. I think I'd just write it as a regular comment next to `U256::one() << 192`.